src/csharp/Grpc.Core/Internal/AsyncCall.cs

```diff
@@ -122,6 +122,9 @@ public TResponse UnaryCall(TRequest msg)
                             catch (Exception e)
                             {
                                 Logger.Error(e, "Exception occurred while invoking completion delegate.");
+                                unaryResponseTcs.SetException(new RpcException(
```

Comment:
I think there are problems with this fix:
-  since in https://github.com/grpc/grpc/issues/29854, ctx.GetReceivedStatusOnClient() throws an exception, the `HandleUnaryResponse()` (which is responsible for cleaning up after the call) never gets invoked, so there is high likelyhood that the call won't get cleaned up properly and it will leak.
- in principle, if the message detail in GetReceivedStatusOnClient can get corrupt (and stop being unmarshallable as UTF-8),  the exact same thing can likely happen for other call types than just syncUnary (e.g. async Unary and potentially some of the streaming ones as well). Rather than fixing the same problem in multiple places, it would be better to make sure we understand why exactly does `ctx.GetReceivedStatusOnClient()` fail and whether that's caused by a bug in C-core or not.